src/16c8e0.c

```diff
@@ -193,11 +200,337 @@ void update_actor_shadows(void) {
+                    paramAmount = 0.0f;
+                }
+                set_screen_overlay_params_back(0, paramAmount);
+            } else {
+                paramAmount += 10.0f;
+                if (paramAmount > 215.0f) {
+                    paramAmount = 215.0f;
+                }
+                set_screen_overlay_params_back(0, paramAmount);
+            }
+        } else if (battleStatus->unk_432 < 0) {
+            paramAmount -= 10.0f;
+            if (paramAmount < 0.0f) {
+                paramAmount = 0.0f;
+                set_screen_overlay_params_back(0xFF, -1.0f);
+                battleStatus->unk_432 = 0;
+            } else {
+                set_screen_overlay_params_back(0, paramAmount);
+            }
+        }
+
+        if (cond || D_802809F6 != -1) {
+            if (D_802809F6 == -1) {
+                if (gGameStatusPtr->demoState == 2) {
+                    u8 paramType;
+                    f32 paramAmount;
+
+                    get_screen_overlay_params(0, &paramType, &paramAmount);
+                    if (paramType == 0xFF) {
+                        D_802809F6 = 0;
+                        set_screen_overlay_params_front(0, 0.0f);
+                    }
+                }
+            } else if (D_802809F6 == 0xFF) {
+                if (gBattleState != 0x23) {
+                    btl_set_state(0x23, 0xFF);
+                }
+            } else {
+                D_802809F6 += 10;
+                if (D_802809F6 > 0xFF) {
+                    D_802809F6 = 0xFF;
+                }
+
+                set_screen_overlay_params_front(0, D_802809F6);
+                set_screen_overlay_color(0, 208, 208, 208);
+                intro_logos_set_fade_alpha(255);
+                intro_logos_set_fade_color(224);
+            }
+        }
+    }
+}
 
 INCLUDE_ASM(s32, "16c8e0", btl_draw_ui);
 
-INCLUDE_ASM(s32, "16c8e0", func_8023ED5C);
+void func_8023ED5C(void) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    RenderTask renderTask;
+    RenderTask* renderTaskPtr = &renderTask;
+    Actor* actor;
+    s32 i;
+
+    if (gBattleState != 0) {
+        func_8024EEA8();
+        if (battleStatus->unk_B8 != NULL) {
+            battleStatus->unk_B8();
+        }
+        if (battleStatus->flags1 & 1) {
+            func_80255FD8();
+
+            if (gCurrentCamID == 1 || gCurrentCamID == 2) {
+                for (i = 0; i < ARRAY_COUNT(battleStatus->enemyActors); i++) {
+                    actor = battleStatus->enemyActors[i];
+
+                    if (actor != NULL && !(actor->flags & 1)) {
+                        renderTaskPtr->appendGfxArg = i;
+                        renderTaskPtr->appendGfx = func_80257B28;
+                        renderTaskPtr->distance = actor->currentPos.z;
+                        renderTaskPtr->renderMode = actor->renderMode;
+                        queue_render_task(renderTaskPtr);
+
+                        if (actor->flags & 0x10000000) {
```

Comment:
Ditto, just add a flag called ACTOR_FLAG_10000000 or something so we can parameterize early pls 👉 👈